mock/mock_proxy.go

```diff
@@ -0,0 +1,280 @@
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mock
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"sync"
+	"testing"
+
+	"github.com/clearcontainers/proxy/api"
+	"github.com/stretchr/testify/assert"
+)
+
+const testContainerid = "123456789"
+const testToken = "testtoken"
+
+type MockProxy struct {
+	t              *testing.T
+	wg             sync.WaitGroup
+	connectionPath string
+
+	// proxy socket
+	listener net.Listener
+
+	// single client to serve
+	cl net.Conn
+
+	ShimConnected chan bool
+
+	lastStdinStream []byte
+
+	Signal           chan ShimSignal
+	ShimDisconnected chan bool
+	StdinReceived    chan bool
+}
+
+func NewMockProxy(t *testing.T, path string) *MockProxy {
+	return &MockProxy{
+		t:                t,
+		connectionPath:   path,
+		lastStdinStream:  nil,
+		Signal:           make(chan ShimSignal, 5),
```

Comment:
We can receive multiple signals from the shim, so preferred to use a buffered channel. 5 seemed a good enough number to test the multiple signals checks that I  plan to add.